projects/packages/stats-admin/src/class-rest-controller.php

```diff
@@ -120,50 +139,101 @@ public function check_user_privileges_callback() {
 	}
 
 	/**
-	 * Resource endpoint.
+	 * Stats resource endpoint.
 	 *
 	 * @param WP_REST_Request $req The request object.
 	 * @return array
 	 */
-	public function get_stats_resource_from_wpcom( $req ) {
-		if ( 'file-downloads' === $req->get_param( 'resource' ) ) {
-			return $this->empty_result();
+	public function get_stats_resource( $req ) {
+		switch ( $req->get_param( 'resource' ) ) {
+			case 'file-downloads':
+				return $this->wpcom_stats->get_file_downloads( $req->get_params() );
+
+			case 'video-plays':
+				return $this->wpcom_stats->get_video_plays( $req->get_params() );
+
+			case 'clicks':
+				return $this->wpcom_stats->get_clicks( $req->get_params() );
+
+			case 'search-terms':
+				return $this->wpcom_stats->get_search_terms( $req->get_params() );
+
+			case 'top-authors':
+				return $this->wpcom_stats->get_top_authors( $req->get_params() );
+
+			case 'country-views':
+				return $this->wpcom_stats->get_views_by_country( $req->get_params() );
+
+			case 'referrers':
+				return $this->wpcom_stats->get_referrers( $req->get_params() );
+
+			case 'top-posts':
+				return $this->wpcom_stats->get_top_posts( $req->get_params() );
+
+			case 'publicize':
+				return $this->wpcom_stats->get_publicize_followers( $req->get_params() );
+
+			case 'followers':
+				return $this->wpcom_stats->get_followers( $req->get_params() );
+
+			case 'tags':
+				return $this->wpcom_stats->get_tags( $req->get_params() );
+
+			case 'visits':
+				return $this->wpcom_stats->get_visits( $req->get_params() );
+
+			case 'comments':
+				return $this->wpcom_stats->get_top_comments( $req->get_params() );
+
+			case 'comment-followers':
+				return $this->wpcom_stats->get_comment_followers( $req->get_params() );
+
+			case 'streak':
+				return $this->wpcom_stats->get_streak( $req->get_params() );
+
+			case 'insights':
+				return $this->wpcom_stats->get_insights( $req->get_params() );
+
+			case 'highlights':
+				return $this->wpcom_stats->get_highlights( $req->get_params() );
+
+			default:
+				return $this->get_forbidden_error();
 		}
-		return static::request_as_blog_cached(
-			sprintf(
-				'/sites/%d/stats/%s?%s',
-				Jetpack_Options::get_option( 'id' ),
-				$req->get_param( 'resource' ),
-				http_build_query(
-					$req->get_params()
-				)
-			),
-			'1.1',
-			array( 'timeout' => 30 )
-		);
+
 	}
 
 	/**
-	 * Sub Resource endpoint e.g. posts/121/likes.
+	 * Return likes of a single post.
 	 *
 	 * @param WP_REST_Request $req The request object.
-	 * @return array
 	 */
-	public function get_site_sub_resource_from_wpcom( $req ) {
-		return static::request_as_blog_cached(
+	public function get_single_post_likes( $req ) {
```

Comment:
Noting this will still not work without the Jetpack plugin present. It's also not very performant as the request will be proxied to WPCOM, then back to the site.